### PR TITLE
coprocessor/* : split dag and old select into independent mod

### PR DESCRIPTION
**Pull request by AndreMouche**
Hi,
     This PR split `dag` and old select into independent mod, @hicqu @BusyJay @andelf PTAL

### CODE REVIEW

File: src/coprocessor/select/topn_heap.rs [hunk 1]
```diff
@@ -0,0 +1,334 @@
+// Copyright 2016 PingCAP, Inc.
```
Comment (BusyJay) [inline src/coprocessor/select/topn_heap.rs]: s/2016/2017/

File: src/coprocessor/select/topn_heap.rs [hunk 2]
```diff
@@ -0,0 +1,334 @@
...
+    }
+}
+
+pub struct TopNHeap {
+    pub rows: BinaryHeap<SortRow>,
+    limit: usize,
+    err: Rc<RefCell<Option<String>>>,
+}
+
+const HEAP_MAX_CAPACITY: usize = 1024;
```
Comment (BusyJay) [inline src/coprocessor/select/topn_heap.rs]: Move the definition to L26.

**Comment by AndreMouche** [comment]
@andelf @hicqu PTAL

**Comment by hicqu** [comment]
LGTM.

**Pull request merged**